File: packages/analysis/src/engine/__tests__/fixtures/sample.ts
import { readFileSync } from "fs";
import { join } from "path";

export interface User {
  id: number;
  name: string;
  email?: string;
}

export type UserStatus = "active" | "inactive" | "pending";

export class UserService {
  private users: User[] = [];

  constructor(private dbPath: string) {
    this.loadUsers();
  }

  async loadUsers(): Promise<void> {
    try {
      const data = readFileSync(join(this.dbPath, "users.json"), "utf8");
      this.users = JSON.parse(data);
    } catch (error) {
      console.error("Failed to load users:", error);
      this.users = [];
    }
  }

  findUser(id: number): User | undefined {
    return this.users.find(user => user.id === id);
  }

  async createUser(userData: Omit<User, "id">): Promise<User> {
    const newUser = {
      id: this.generateId(),
      ...userData
    };

    this.users.push(newUser);
    await this.saveUsers();
    return newUser;
  }

  private generateId(): number {
    return Math.max(...this.users.map(u => u.id), 0) + 1;
  }

  private async saveUsers(): Promise<void> {
    const data = JSON.stringify(this.users, null, 2);
    // writeFileSync would be called here in real implementation
    console.log("Saving users data...");
  }
}

export function validateEmail(email: string): boolean {
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email);
}

export default UserService;

File: packages/analysis/src/engine/__tests__/fixtures/sample.rs
use std::collections::HashMap;

pub struct User {
    pub id: u32,
    pub name: String,
    pub email: Option<String>,
}

pub enum UserStatus {
    Active,
    Inactive,
    Pending,
}

impl User {
    pub fn new(id: u32, name: String) -> Self {
        User {
            id,
            name,
            email: None,
        }
    }

    pub fn set_email(&mut self, email: String) {
        self.email = Some(email);
    }

    pub fn is_valid(&self) -> bool {
        !self.name.is_empty() && self.id > 0
    }
}

pub fn validate_email(email: &str) -> bool {
    email.contains('@') && email.contains('.')
}

pub fn create_user_map() -> HashMap<u32, User> {
    let mut users = HashMap::new();

    let user1 = User::new(1, "Alice".to_string());
    let user2 = User::new(2, "Bob".to_string());

    users.insert(user1.id, user1);
    users.insert(user2.id, user2);

    users
}

fn helper_function() -> String {
    "helper".to_string()
}
